### PR TITLE
Add pipenv and update asdf to latest

### PR DESCRIPTION
**Pull request by RothAndrew**
## what/why

- Update asdf to latest. This fixes an issue with the Go asdf plugin.
- Add pipenv so it can be used by python projects.

## Notes

- I initially also wanted to install pipx, since that is how the pipenv docs recommend it should be installed, but I ran into issues. Since this container is "meant" to run under the user `anvil` the `anvil` user is the one that does all the asdf installs in the Dockerfile. However, since this container is frequently run as `root`, some hacky bits had to be added to the bottom of the dockerfile to make that work. This works fine with things installed to asdf's bin directory, but it gets messed up when you try to do things like `pip install --user` since that is installed to a different directory. Because of this, I left it how it was and didn't install pipx. The side effect is if you change the version of python in a `.tool-versions` file you won't have the tools that were installed with pip like `aws`, `pre-commit`, or `pipenv`. We should look more closely at this later, as this MR isn't introducing this issue, but it may be surfacing it more.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+nodejs 12.18.3
```
Comment (BradStaton) [inline .tool-versions:1]: Why is this necessary for `pip`?

**Review by BradStaton — COMMENTED** [review]
Should you also add a test (pip_spec.sh) to verify that pip is installed?

**Comment by RothAndrew** [comment]
> Should you also add a test (pip_spec.sh) to verify that pip is installed?

Yes, thanks. I forgot.

File: .tool-versions (hunk 1 above)
Comment (RothAndrew) [reply .tool-versions:1]: the prettier pre-commit hook

File: src/docker/Dockerfile [hunk 2]
```diff
@@ -132,7 +132,7 @@ USER anvil
 WORKDIR /home/anvil
 
 # Install asdf. Get versions from https://github.com/asdf-vm/asdf/releases
-ARG ASDF_VERSION="0.7.8"
+ARG ASDF_VERSION="0.8.0"
```
Comment (bstascavage) [inline src/docker/Dockerfile:135]: Question I've wonder about this:

Do we want to pull all of these binary versions out into one central file ? I noticed that they are scattered around the `Dockerfile` a bit.

**Review by saic-statonb — APPROVED** [review]
LGTM

File: src/docker/Dockerfile (hunk 2 above)
Comment (RothAndrew) [reply src/docker/Dockerfile:135]: They live next to where they are needed so the docker layer cache gets busted less often. If they were all in one file changing one version would bust the whole cache even if it were all the way at the bottom of the dockerfile

Comment (bstascavage) [reply src/docker/Dockerfile:135]: Yeah I had a feeling that was the reason.  Thanks for confirming.
